### DISCUSSION TITLE
Подскажите, в какую сторону копать (Keenetic Giga II)

### DISCUSSION THREAD

**Discussion by DarkReggae** [Anonym-tsk/nfqws-keenetic#105] [Q&A] [answered]
Роутер Keenetic Giga II чёрный, прошивка 2.16.D.12.0-11 (последняя на данный момент).
До недавнего времени NFQWS-Keenetic работал превосходно, потом, сначала перестал работать с устройствами, подключенными по WiFi, работал только по LAN. После попытки обновиться перестал работать совсем. Полная переустановка всего с нуля результатов не дала.

При этом NFQWS вполне работает с компьютера из-под Win10 (в составе пакета zapret-win-bundle от bol-van).

Конфигурация одинаковая, на основе данных blockcheck из того же пакета. 

Вывод NFQWS под Windows:

> we have 5 user defined desync profile(s) and default low priority profile 0
> Loading hostlist C:\Program Files\Zapret-WIN\zapret-winws\list-youtube.txt
> loading plain text list
> Loaded 18 hosts from C:\Program Files\Zapret-WIN\zapret-winws\list-youtube.txt
> Loading hostlist C:\Program Files\Zapret-WIN\zapret-winws\list-youtube.txt
> loading plain text list
> Loaded 18 hosts from C:\Program Files\Zapret-WIN\zapret-winws\list-youtube.txt
> windivert initialized. capture is started.
> 
> C:\Windows\system32>curl --connect-to ::speedtest.selectel.ru https://manifest.googlevideo.com/100MB -k -o/nul
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
> 100  100M  100  100M    0     0  21.2M      0  0:00:04  0:00:04 --:--:-- 21.2M
> C:\Windows\system32>

Конфигурация NFQWS под Windows (найдена blockcheck'ом из того же пакета):

> start "zapret: http,https,quic" /min "%~dp0winws.exe" ^
> --wf-tcp=80,443 --wf-udp=443 ^
> --filter-udp=443 --hostlist="%~dp0list-youtube.txt" --dpi-desync=fake --dpi-desync-repeats=16 --dpi-desync-fake-quic="%~dp0quic_initial_www_google_com.bin" --new ^
> --filter-udp=443 --dpi-desync=fake --dpi-desync-repeats=16 --new ^
> --filter-tcp=80 --dpi-desync=fake --dpi-desync-ttl=3 --new ^
> --filter-tcp=443 --hostlist="%~dp0list-youtube.txt" --dpi-desync=fake,disorder2 --dpi-desync-ttl=3 --new ^
> --dpi-desync=fake,disorder2 --dpi-desync-ttl=3 --dpi-desync-fooling=badseq,md5sig

Вывод NFQWS-Keenetic на роутере:

> ~ # /opt/etc/init.d/S51nfqws start
> Cannot find xt_multiport.ko module
> we have 6 user defined desync profile(s) and default low priority profile 0
> Loading hostlist /opt/etc/nfqws/auto.list
> loading plain text list
> Loaded 1 hosts from /opt/etc/nfqws/auto.list
> Loading hostlist /opt/etc/nfqws/user.list
> loading plain text list
> Loaded 30 hosts from /opt/etc/nfqws/user.list
> Loading hostlist /opt/etc/nfqws/auto.list
> loading plain text list
> Loaded 1 hosts from /opt/etc/nfqws/auto.list
> Loading hostlist /opt/etc/nfqws/user.list
> loading plain text list
> Loaded 30 hosts from /opt/etc/nfqws/user.list
> Loading hostlist /opt/etc/nfqws/exclude.list
> loading plain text list
> Loaded 28 hosts from /opt/etc/nfqws/exclude.list
> Loading hostlist /opt/etc/nfqws/exclude.list
> loading plain text list
> Loaded 28 hosts from /opt/etc/nfqws/exclude.list
> Started NFQWS service
> ~ # curl --connect-to ::speedtest.selectel.ru https://manifest.googlevideo.com/100MB -k -o/dev/null
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                  Dload  Upload   Total   Spent    Left  Speed
>   0     0    0     0    0     0      0      0 --:--:--  0:00:09 --:--:--     0^C
> ~ #

Конфиг NFQWS-Keenetic на роутере (закомментированные строки убрал):

> ISP_INTERFACE="eth2.2"
> NFQWS_ARGS="--dpi-desync=fake,disorder2 --dpi-desync-ttl=3 --new"
> NFQWS_ARGS_QUIC="--dpi-desync=fake --dpi-desync-repeats=16 --dpi-desync-fake-quic=/opt/etc/nfqws/quic_initial_www_google_com.bin --new"
> NFQWS_ARGS_UDP="--filter-udp=50000-50099 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-repeats=16 --new"
> NFQWS_EXTRA_ARGS="--hostlist=/opt/etc/nfqws/user.list --hostlist-auto=/opt/etc/nfqws/auto.list --hostlist-auto-debug=/opt/var/log/nfqws.log --hostlist-exclude=/opt/etc/nfqws/exclude.list"
> IPV6_ENABLED=1
> TCP_PORTS=443
> UDP_PORTS=443,50000:50099
> LOG_LEVEL=0
> NFQUEUE_NUM=200
> USER=nobody
> CONFIG_VERSION=4

Подскажите, в какую сторону копать?

**Comment by DarkReggae** [comment]
UPD: Откат на версию NFQWS-Keenetic 2.3.3 ничего не дал. Полную перепрошивку роутера с обнулением всего ещё не пробовал.

**Comment by bear-gha** [comment]
> сначала перестал работать с устройствами, подключенными по WiFi

надо ковырять что в этот момент поменялось, т.к. если работа была по LAN в этот же момент, то дело не в "обходе".

> ISP_INTERFACE="eth2.2"

оно точно там?

**Reply by DarkReggae** [reply]
Точно, ifconfig показывает внешний айпишник там. Впрочем, пробовал вписывать туда поочерёдно все интерфейсы из списка, эффект нулевой.

**Comment by kotovasia5120** [comment]
>  сначала перестал работать с устройствами, подключенными по WiFi, работал только по LAN.

у меня на giga II при подключении моих телефонов wifi отваливается у всех. 
Если подключатся с ноутбука или steamdeck то wifi работает без сбоев.
Не знаю чем ему телефоны не нравятся

прошивка 2.16.D.12.0-8 вроде

**Comment by Anonym-tsk** [comment, marked as answer]
> NFQWS_ARGS="--dpi-desync=fake,disorder2 --dpi-desync-ttl=3 --new"
> NFQWS_ARGS_QUIC="--dpi-desync=fake --dpi-desync-repeats=16 --dpi-desync-fake-quic=/opt/etc/nfqws/quic_initial_www_google_com.bin --new"
> NFQWS_ARGS_UDP="--filter-udp=50000-50099 --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-repeats=16 --new"

Верните конфиг по-умолчанию и проверьте. У вас тут очень криво всё

**Reply by DarkReggae** [reply]
Так я, собственно, и начал ковыряться потому что с конфигом по умолчанию оно не работает, иначе зачем бы полез вообще ))
Вот только что поставилась версия 2.5.9, ничего не делал, всё по умолчанию, но увы ((

**Comment by DarkReggae** [comment]
После полной переустановки прошивки роутера NFQWS-Keenetic заработал как положено с дефолтными настройками, при подключении как по LAN, так и по WiFi всё прекрасно. Проблема была где-то в прошивке роутера, всем спасибо за ответы, тему можно закрывать.
